pandas/core/arrays/interval.py

```diff
@@ -102,6 +105,7 @@
 
 
 IntervalArrayT = TypeVar("IntervalArrayT", bound="IntervalArray")
+IntervalSideT = Union[TimeArrayLike, np.ndarray]
```

Comment:
nice.  an option if you want to be more specific can specify `npt.NDArray[numeric_dtypes]`.  OK to call this out of scope

Reply:
I can do it in a follow-up? I think I'd prefer an iterative approach, i.e. closing the book on this method cleaning PR and picking up the typing issue as a new PR.